src/url.js

```diff
@@ -71,7 +69,7 @@ export function parseUrl(url) {
 
   const fromCache = cache[url];
   if (fromCache) {
-    return fromCache;
+    return Object.assign({}, fromCache);
```

Comment:
This is not what I thought the plan was. Where is the code where we mutate the cache contents?